mfe_test.go

```diff
@@ -0,0 +1,1270 @@
+		Interior loop (2475,2485) CG; (2476,2484) CG:  -330
+		Hairpin  loop (2476,2484) CG              :   370
+		Interior loop (2504,2519) UA; (2505,2518) GC:  -210
+		Interior loop (2505,2518) GC; (2506,2517) UA:  -220
+		Interior loop (2506,2517) UA; (2507,2516) AU:  -130
+		Interior loop (2507,2516) AU; (2508,2515) GC:  -210
+		Hairpin  loop (2508,2515) GC              :   470
+		Multi    loop (2447,2523) GC              :   360
+		Interior loop (2549,2626) GC; (2550,2625) GU:  -150
+		Interior loop (2550,2625) GU; (2551,2624) CG:  -250
+		Interior loop (2551,2624) CG; (2552,2623) UG:  -210
+		Interior loop (2552,2623) UG; (2553,2622) GC:  -140
+		Interior loop (2553,2622) GC; (2554,2621) CG:  -340
+		Interior loop (2554,2621) CG; (2555,2620) UA:  -210
+		Interior loop (2555,2620) UA; (2556,2619) GC:  -210
+		Interior loop (2556,2619) GC; (2557,2618) CG:  -340
+		Interior loop (2562,2572) GC; (2563,2571) GU:  -150
+		Interior loop (2563,2571) GU; (2564,2570) CG:  -250
+		Hairpin  loop (2564,2570) CG              :   340
+		Interior loop (2575,2617) GC; (2576,2616) UG:  -250
+		Interior loop (2576,2616) UG; (2577,2615) CG:  -150
+		Interior loop (2577,2615) CG; (2578,2614) UA:  -210
+		Interior loop (2578,2614) UA; (2579,2613) UA:   -90
+		Interior loop (2579,2613) UA; (2580,2612) AU:  -130
+		Interior loop (2580,2612) AU; (2581,2610) CG:   160
+		Interior loop (2581,2610) CG; (2582,2609) CG:  -330
+		Interior loop (2582,2609) CG; (2583,2608) GC:  -240
+		Interior loop (2583,2608) GC; (2584,2607) GC:  -330
+		Interior loop (2584,2607) GC; (2585,2601) GU:   450
+		Interior loop (2585,2601) GU; (2586,2600) UA:  -140
+		Interior loop (2586,2600) UA; (2587,2599) UG:  -130
+		Interior loop (2587,2599) UG; (2588,2598) GC:  -140
+		Hairpin  loop (2588,2598) GC              :   390
+		Multi    loop (2557,2618) CG              :   250
+		Interior loop (2627,2657) GC; (2628,2656) GC:  -330
+		Interior loop (2628,2656) GC; (2629,2655) GC:  -330
+		Interior loop (2629,2655) GC; (2630,2654) CG:  -340
+		Interior loop (2630,2654) CG; (2631,2653) UA:  -210
+		Interior loop (2631,2653) UA; (2632,2652) GC:  -210
+		Interior loop (2632,2652) GC; (2634,2646) AU:   510
+		Interior loop (2634,2646) AU; (2635,2645) CG:  -220
+		Interior loop (2635,2645) CG; (2636,2644) GC:  -240
+		Interior loop (2636,2644) GC; (2637,2643) GU:  -150
+		Hairpin  loop (2637,2643) GU              :   520
+		Multi    loop (2275,2662) AU              :  -430
+		-936.5`, t)
+}
+
+func test(sequence, structure string, expected_output string, t *testing.T) {
+	output := captureOutput(func() {
+		mfe, energyContributions, _ := CalculateMFE(sequence, structure, DefaultTemperature)
+		logEnergyContributions(energyContributions, sequence)
+		log.Printf("%v", mfe)
+	})
+
+	if stripWhiteSpace(output) != stripWhiteSpace(expected_output) {
+		t.Errorf("\n\nFailed to calcualte mfe for '%v'. \nExpected: \n'%v'\n\nGot: \n'%v'\n\n",
+			sequence, expected_output, output)
+	}
+}
```

Comment:
Lets also include an example of a realistic function that a user might use.

Reply:
I added the test `ExampleMinimumFreeEnergy()`. Could you confirm that it's a realistic example?